### ISSUE TITLE
LossNet  model for custom dataset

### ISSUE THREAD

**Issue by kanzaali** [euphoria0-0/Learning-Loss-for-Active-Learning#13]
Hi, Thank you so much for your code.
I am unable to use the LossNet  model for domain-specific datasets with different dimensions. It always says is forward()

Input dimension should be at least 3

or ndexError: Dimension out of range (expected to be in range of [-1, 0], but got -3)

**Comment by euphoria0-0** [comment]
What is your task?
The inputs for LossNet are the intermediate features from task model(e.g. resnet).

**Comment by kanzaali** [comment]
I only want to check the performance on a different dataset which has 64 sizes, 3 channels, 4 classes, and batch size 32 but LossNet is giving errors. I am not getting how to change ResNet.

**Comment by euphoria0-0** [comment]
Then probabily you can change the resize factor.
The default dimensions of LossNet will be matched.

**Comment by kanzaali** [comment]
resize factor in the data loader?

**Comment by euphoria0-0** [comment]
In dataset : ) you can use transforms

**Comment by manza-ari** [comment]
yeah I am already doing it 

`
def load_dataset(dataset):
    train_transform = T.Compose([
        T.Resize((64,64)),
        T.RandomHorizontalFlip(),
        T.RandomCrop(size=64, padding = 4),    #224 for COVID and 64 for Satellite 
        T.ToTensor(),
        T.Normalize([0.4914, 0.4822, 0.4465], [0.2023, 0.1994, 0.2010]) # T.Normalize((0.5071, 0.4867, 0.4408), (0.2675, 0.2565, 0.2761)) # CIFAR-100
    ])

    test_transform = T.Compose([
        T.ToTensor(),
        T.Normalize([0.4914, 0.4822, 0.4465], [0.2023, 0.1994, 0.2010]) # T.Normalize((0.5071, 0.4867, 0.4408), (0.2675, 0.2565, 0.2761)) # CIFAR-100
    ])
    if dataset == 'Satellite':
`

**Comment by manza-ari** [comment]
I am already doing it 
`
def load_dataset(dataset):
    train_transform = T.Compose([
        T.Resize((64,64)),
        T.RandomHorizontalFlip(),
        T.RandomCrop(size=64, padding = 4),    
        T.ToTensor(),
        T.Normalize([0.4914, 0.4822, 0.4465], [0.2023, 0.1994, 0.2010]) # T.Normalize((0.5071, 0.4867, 0.4408), (0.2675, 0.2565, 0.2761)) # CIFAR-100
    ])

    test_transform = T.Compose([
        T.ToTensor(),
        T.Normalize([0.4914, 0.4822, 0.4465], [0.2023, 0.1994, 0.2010]) 
    ])
    if dataset == 'Satellite':
`

**Comment by euphoria0-0** [comment]
You should modify the feature dimensions for ResNet in image classification cass.
https://discuss.pytorch.org/t/altering-resnet18-for-single-channel-images/29198/3
It will work probably if you change the first convolution layer in ResNet as `self.conv1 = nn.Conv2d(3, 64, kernel_size=7, stride=2, padding=3, bias=False)`.
